units/helpers_test.go

```diff
@@ -24,11 +27,34 @@ func TestReadJournalOutputSince_Linux(t *testing.T) {
 		t.Skip()
 	}
 
-	r, err := ReadJournalOutputSince("not-existing.service", "")
+	ctxDeadline := time.Now().Add(1 * time.Hour) // this test shouldn't take longer than an hour, should it?
+	ctx, cancel := context.WithDeadline(context.Background(), ctxDeadline)
+	defer cancel()
+	r, err := ReadJournalOutputSince(ctx, "not-existing.service", "")
 	require.NoError(t, err)
 
 	data, err := ioutil.ReadAll(r)
 	require.NoError(t, err)
 
 	assert.Empty(t, data)
+
+	assert.IsType(t, &TimeoutReadCloser{}, r)
+	tr := r.(*TimeoutReadCloser)
+	deadline, ok := tr.ctx.Deadline()
+	assert.True(t, ok, "Context expected to have a deadline but it had none")
+	assert.Equal(t, ctxDeadline, deadline, "Deadline has unexpected value")
```

Comment:
What are you testing here?

Reply:
We want to make sure that `ReadJournalOutputSince` creates a `TimeoutReadCloser` wrapper for the underlying systemd `ReadCloser` with the deadline that we provided when calling `ReadJournalOutputSince`.